src/components/js/EcAdminSettingsForm.js

```diff
@@ -191,49 +193,110 @@ export default {
           data.forEach(row => {
-            const parsedData = {}
+            const parsedData = {};
             for (const head in row) {
               if (row[head]) {
-                const field = head.replace(/\w+\(([^)]+)\)/i, '$1')
+                const field = head.replace(/\w+\(([^)]+)\)/i, '$1');
                 const value = head.startsWith('Number')
                   ? Number(row[head])
                   : head.startsWith('Boolean')
                     ? Boolean(row[head] && !row[head].toUpperCase().startsWith('FALS'))
-                    : row[head]
-                const fields = field.split(/[.[\]]/)
+                    : row[head];
+                const fields = field.split(/[.[\]]/);
                 if (fields.length > 1) {
-                  let nestedField = parsedData
+                  let currentField = parsedData;
                   for (let i = 0; i < fields.length - 1; i++) {
-                    if (!nestedField[fields[i]]) {
-                      nestedField[fields[i]] = {}
+                    const currentFieldName = fields[i];
+                    if (!currentField[currentFieldName]) {
+                      // Check if the next part of the field is a numeric index
+                      const isArrayIndex = /^\d+$/.test(fields[i + 1]);
+                      if (isArrayIndex) {
+                        currentField[currentFieldName] = [];
+                      } else {
+                        currentField[currentFieldName] = {};
+                      }
                     }
-                    nestedField = nestedField[fields[i]]
+                    currentField = currentField[currentFieldName];
+                  }
+                  const lastFieldName = fields[fields.length - 1];
+                  if (/^\d+$/.test(lastFieldName)) {
+                    // It's a numeric index, insert into array
+                    currentField.push(value);
+                  } else {
+                    currentField[lastFieldName] = value;
                   }
-                  nestedField[fields[fields.length - 1]] = value
                 } else {
-                  parsedData[field] = value
+                  parsedData[field] = value;
                 }
               }
             }
             if (Object.keys(parsedData).length) {
-              dataList.push(parsedData)
+              dataList.push(parsedData);
             }
-          })
+          });
+        },
+      });
+      return false;
+    },
+
+    downloadCsv (dataList) {
+      const parseDocToRow = doc => {
+        const row = dot.dot(doc)
+        for (const field in row) {
+          if (row[field] !== undefined) {
+            const type = typeof row[field]
+            if (type !== 'object') {
+              // save var type on row header
+              let fixedField = field
+              if (/\[(\d+)\]/.test(field)) {
+                fixedField = field.replace(/\[(\d+)\]/, '.$1')
+              }
```

Comment:
Uai não vejo porque isso aqui não...

Reply:
A intenção é corrigir tabela mal formatada? Isso aqui parece um bug na real

Reply:
Eu fiz isso ai pra seguir o padrão, no painel administrativo, o array é String(variations[0]._id) por exemplo e ai, seria dessa forma também, viria String(product_ids[0]), o problema é só as tabelas que já foram feitas e estão na mão dos clientes, na verdade, elas nem tem String(...), é só product_ids.0, product_ids.1, dai fiquei sem saber qual seguir, acabei seguindo as tabelas que já estão rodando pelos clientes. Mas se for o caso, posso mudar essas tabelas também